crud.py

```diff
@@ -0,0 +1,71 @@
+from sqlalchemy.orm import Session
+
+from db import models
+import schemas
+
+
+def get_book_list(db: Session, author_id: str | None = None):
+    queryset = db.query(models.DBBook)
+
+    if author_id:
+        queryset = queryset.filter(models.DBBook.author_id == author_id)
+
+    return queryset
+
+
+def get_book(db: Session, book_id: int):
+    return db.query(models.DBBook).filter(models.DBBook.id == book_id).first()
+
+
+def get_book_by_title(db: Session, title: str):
+    return db.query(models.DBBook).filter(models.DBBook.title == title).first()
+
+
+def create_book(db: Session, book: schemas.BookCreate):
+    db_book = models.DBBook(
+        title=book.title,
+        summary=book.summary,
+        publication_date=book.publication_date,
+        author_id=book.author_id,
+    )
+    db.add(db_book)
+    db.commit()
+    db.refresh(db_book)
+
+    return db_book
+
+
+def get_all_authors(db: Session):
+    return db.query(models.DBAuthor)
```

Comment:
This function should return a list of authors. Use .all() to fetch the results and add appropriate type hinting.